admin/class-yoast-dashboard-widget.php

```diff
@@ -110,48 +110,50 @@ private function set_statistic_items_for_this_user( $transient ) {
 	 * @return array
 	 */
 	private function get_seo_scores_with_post_count() {
+		$ranks = WPSEO_Rank::get_all_ranks();
+
+		return array_map( array( $this, 'map_rank_to_widget' ), $ranks );
+	}
+
+	/**
+	 * Converts a rank to data usable in the dashboard widget
+	 *
+	 * @param WPSEO_Rank $rank The rank to map.
+	 *
+	 * @return array
+	 */
+	private function map_rank_to_widget( $rank ) {
 		return array(
-			array(
-				'seo_rank' => 'good',
-				'title'    => __( 'Posts with good SEO score', 'wordpress-seo' ),
-				'class'    => 'wpseo-glance-good',
-				'count'    => $this->statistics->get_good_seo_post_count(),
-			),
-			array(
-				'seo_rank' => 'ok',
-				'title'    => __( 'Posts with OK SEO score', 'wordpress-seo' ),
-				'class'    => 'wpseo-glance-ok',
-				'count'    => $this->statistics->get_ok_seo_post_count(),
-			),
-			array(
-				'seo_rank' => 'poor',
-				'title'    => __( 'Posts with poor SEO score', 'wordpress-seo' ),
-				'class'    => 'wpseo-glance-poor',
-				'count'    => $this->statistics->get_poor_seo_post_count(),
-			),
-			array(
-				'seo_rank' => 'bad',
-				'title'    => __( 'Posts with bad SEO score', 'wordpress-seo' ),
-				'class'    => 'wpseo-glance-bad',
-				'count'    => $this->statistics->get_bad_seo_post_count(),
-			),
-			array(
-				'seo_rank' => 'na',
-				'title'    => __( 'Posts without focus keyword', 'wordpress-seo' ),
-				'class'    => 'wpseo-glance-na',
-				'count'    => $this->statistics->get_no_focus_post_count(),
-			),
-			array(
-				'seo_rank' => 'noindex',
-				/* translators: %s expands to <code>noindex</code> */
-				'title'    => sprintf( __( 'Posts that are set to %s', 'wordpress-seo' ), '<code>noindex</code>' ),
-				'class'    => 'wpseo-glance-noindex',
-				'count'    => $this->statistics->get_no_index_post_count(),
-			),
+			'seo_rank'   => $rank->get_rank(),
+			'title'      => $this->get_title_for_rank( $rank ),
+			'class'      => 'wpseo-glance-' . $rank->get_css_class(),
+			'icon_class' => $rank->get_css_class(),
+			'count'      => $this->statistics->get_post_count( $rank ),
 		);
 	}
 
 	/**
+	 * Returns a dashboard widget label to use for a certain rank
+	 *
+	 * @param WPSEO_Rank $rank The rank to return a label for.
+	 *
+	 * @return string
+	 */
+	private function get_title_for_rank( WPSEO_Rank $rank ) {
```

Comment:
This is so we can verify that we have a correct WPSEO_Rank and we can typehint the rank.